[samr21] Updates

al1img opened the pull request:
This pull request addresses following samr21 related issues:

1. Fixes gcc 7.3.1 compilation error: https://github.com/openthread/openthread/issues/2967
2. Add possibility to define other SAMR21 boards with CLI
3. Update README.md
4. Use HW AES-128 module as mbestls now supports 128 bit random generator: https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/openthread-users/FXv41nDe5kY/VFl3R0AsBgAJ
5. Set RF state to OT_RADIO_STATE_RECEIVE when TX finished: https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/openthread-users/XZ8FU6kmToM/GvFT3pMIBwAJ
6. Pass ACK frame with pending indication to otPlatRadioTxDone
7. Add otPlatLog implementation to use Microchip DGI over SPI (http://ww1.microchip.com/downloads/en/devicedoc/40001905b.pdf).

bukepo commented on examples/platforms/samr21/crypto/aes_alt.c:
```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2017, The OpenThread Authors.
```
2017 -> 2019?

bukepo commented on examples/platforms/samr21/crypto/aes_alt.h:
```diff
@@ -0,0 +1,165 @@
+/*
+ *  Copyright (c) 2017, The OpenThread Authors.
```
ditto

bukepo commented on examples/platforms/samr21/crypto/samr21-mbedtls-config.h:
```diff
@@ -0,0 +1,47 @@
+/*
+ *  Copyright (c) 2017, The OpenThread Authors.
```
ditto

jwhui commented:
@al1img, thanks for these updates!

Can you also take a look at making `SETTINGS_CONFIG_PAGE_NUM` to be 2 or larger? The flash driver requires two sectors so that it can ensure no settings are lost even during unexpected reboots. #3416 is currently blocking on this.

al1img commented:
@jwhui, I've allocated 4k for NVRAM. The page size is 256 bytes (actually 64 bytes, but erase operation is done on a row which is four pages). So, currently SETTINGS_CONFIG_PAGE_NUM = 16 with SETTINGS_CONFIG_PAGE_SIZE = 256. Is it ok for #3416 issue?

jwhui commented:
I believe a `SETTINGS_CONFIG_PAGE_SIZE  = 256` and `SETTINGS_CONFIG_PAGE_NUM = 16` should be sufficient. However, the build check for #3416 currently fails for what's in master (see [travis log](https://travis-ci.org/openthread/openthread/jobs/484179806)). Thoughts?

al1img commented:
@jwhui, it fails due to `SETTINGS_CONFIG_PAGE_NUM` is defined as `(((uint32_t)&__d_nv_mem_end - (uint32_t)&__d_nv_mem_start) / SETTINGS_CONFIG_PAGE_SIZE)` where `__d_nv_mem_start` and `__d_nv_mem_end` are defined in the linker script. That's why `#if (SETTINGS_CONFIG_PAGE_NUM <= 1)` doesn't work foe SAMR21. I've added new commit which defines `SETTINGS_CONFIG_PAGE_SIZE` as constant and put explanation comment.
I've also updated copyright year to 2019 in new files.
